integration/profiles_gateway_test.go

```diff
@@ -0,0 +1,333 @@
+		},
+		{
+			name:   "success response",
+			json:   readJSON.GetPath("success"),
+			expect: `{"code":"OK","status":200}`,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ValidateJSONSchema(t, test.json, test.expect)
+		})
+	}
+}
+
+// TestUpdateProfile_gateway uses the REST gateway to create a new profile and
+// then read that profile from the application by ID
+// 1. Create a profile with a POST request to /profiles
+// 2. Use the corresponding ID to GET the profile at /profiles
+// 3. Create a separate profile with updated fields
+// 4. Send a PUT request to /profiles/ID with the updated fields
+// 5. Ensure the response from PUT has expected fields
+func TestUpdateProfile_gateway(t *testing.T) {
+	dbTest.Reset(t)
+	profile := pb.Profile{
+		Name: "photography", Notes: "profile to show my photography portfolio",
+	}
+	resCreate, err := MakeRequestWithDefaults(
+		http.MethodPost,
+		"http://localhost:8080/v1/profiles",
+		profile,
+	)
+	if err != nil {
+		t.Fatalf("unable to create profile: %v", err)
+	}
+	createJSON, err := simplejson.NewFromReader(resCreate.Body)
+	if err != nil {
+		t.Fatalf("unable to unmarshal json response: %v", err)
+	}
+	id, err := createJSON.GetPath("result", "id").String()
+	if err != nil {
+		t.Fatalf("unable to get profile id from json response: %v", err)
+	}
+	id = strings.TrimPrefix(id, fmt.Sprintf("%s/%s/", cmd.ApplicationID, "profiles"))
+	updated := pb.Profile{
+		Name:  "woodworking",
+		Notes: "profile to show my woodworking portfolio",
+	}
+	resUpdate, err := MakeRequestWithDefaults(
+		http.MethodPut, fmt.Sprintf("http://localhost:8080/v1/profiles/%s", id),
+		updated,
+	)
+	if err != nil {
+		t.Fatalf("unable to update profile: %v", err)
+	}
+	updateJSON, err := simplejson.NewFromReader(resUpdate.Body)
+	if err != nil {
+		t.Fatalf("unable to unmarshal json response: %v", err)
+	}
+	var tests = []struct {
+		name   string
+		json   *simplejson.Json
+		expect string
+	}{
+		{
+			name:   "profile id",
+			json:   updateJSON.GetPath("result", "id"),
+			expect: `"atlas-contacts-app/profiles/1"`,
+		},
+		{
+			name:   "profile notes",
+			json:   updateJSON.GetPath("result", "notes"),
+			expect: `"profile to show my woodworking portfolio"`,
+		},
+		{
+			name:   "profile name",
+			json:   updateJSON.GetPath("result", "name"),
+			expect: `"woodworking"`,
+		},
+		{
+			name:   "success response",
+			json:   updateJSON.GetPath("success"),
+			expect: `{"code":"OK","status":200}`,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ValidateJSONSchema(t, test.json, test.expect)
+		})
+	}
+}
+
+// TestDeleteProfile_gateway uses the REST gateway to create a new profile and
+// ensures it can get deleted by sending a DELETE request to /profiles/{id}
+// 1. Create a profile entry with a POST request
+// 2. Delete the profile by sending a DELETE request to /profiles/{id}
+// 3. Ensure the DELETE response JSON matches the expected schema
```

Comment:
you could additionally do `4. Ensure a following GET request to that id returns a 404`